test/torchaudio_unittest/models/tacotron2/model_test.py

```diff
@@ -0,0 +1,202 @@
+        out = model(x, input_lengths)
+
+        assert out.size() == (n_batch, n_seq, encoder_embedding_dim)
+
+    @parameterized.expand([(32, 64, 512, torch.float32, torch.device('cpu'))])
+    def test_cpu_encoder_output(self, n_batch, n_seq, encoder_embedding_dim, dtype, device):
+        model = self._get_model(encoder_embedding_dim=encoder_embedding_dim).to(device).eval()
+        self._output_shape_test(model, n_batch, encoder_embedding_dim, n_seq, device, dtype)
+
+    @parameterized.expand([(32, 64, 512, torch.float32, torch.device('cuda'))])
+    @skipIfNoCuda
+    def test_gpu_encoder_output(self, n_batch, n_seq, encoder_embedding_dim, dtype, device):
+        model = self._get_model(encoder_embedding_dim=encoder_embedding_dim).to(device).eval()
+        self._output_shape_test(model, n_batch, encoder_embedding_dim, n_seq, device, dtype)
+
+
+class TestDecoder(TorchaudioTestCase, TorchscriptConsistencyMixin):
+
+    def _get_model(self, n_mels=80, encoder_embedding_dim=512):
+        model = _Decoder(n_mels=n_mels,
+                         n_frames_per_step=1,
+                         encoder_embedding_dim=encoder_embedding_dim,
+                         attention_dim=128,
+                         attention_rnn_dim=1024,
+                         attention_location_n_filters=32,
+                         attention_location_kernel_size=31,
+                         decoder_rnn_dim=1024,
+                         prenet_dim=256,
+                         max_decoder_steps=2000,
+                         gate_threshold=0.5,
+                         p_attention_dropout=0.1,
+                         p_decoder_dropout=0.1,
+                         early_stopping=False)
+        return model
+
+    def _torchscript_consistency_test(self,
+                                      model,
+                                      n_mels,
+                                      n_batch,
+                                      encoder_embedding_dim,
+                                      n_seq,
+                                      device,
+                                      dtype):
+        """Validate the torchscript consistency of a Decoder.
+        """
+
+        n_time_steps = 200
+
+        memory = torch.rand(n_batch, n_seq, encoder_embedding_dim, dtype=dtype, device=device)
+        decoder_inputs = torch.rand(n_batch, n_mels, n_time_steps, dtype=dtype, device=device)
+        memory_lengths = torch.ones(n_batch, dtype=torch.int32, device=device)
+
+        self._assert_torchscript_consistency(model, (memory, decoder_inputs, memory_lengths))
+
+    def _output_shape_test(self,
+                           model,
+                           n_mels,
+                           n_batch,
+                           encoder_embedding_dim,
+                           n_seq,
+                           device,
+                           dtype):
+        """Validate the output dimensions of a Decoder.
```

Comment:
Can you replace "validate" with something more descriptive?

What conditions output dimensions should be satisfying?

Reply:
I've changed it to 
```python
        """Feed tensors with specific shape into Tacotron2 Decoder and validate
        that it outputs with a tensor with expected shape.
        """
```
Please let me know if this is clear enough or more detailed comments should be written.
Thanks.